detection_rules/custom_schemas.py

```diff
@@ -37,3 +39,77 @@ def get_custom_schemas(stack_version: str = None) -> dict:
                     raise ValueError(f"Custom schema must be a file: {schema_path}")
 
     return custom_schema_dump
+
+
+def resolve_schema_path(path: str) -> Path:
+    """Helper function to resolve the schema path."""
+    path_obj = Path(path)
+    return path_obj if path_obj.is_absolute() else RULES_CONFIG.stack_schema_map_file.parent.joinpath(path)
+
+
+def update_data(index: str, field: str, data: dict) -> dict:
+    """Update the schema entry with the appropriate index and field."""
+    if index not in data:
+        data[index] = {}
+    data[index][field] = "keyword"
+    return data
+
+
+def update_stack_schema_map(stack_schema_map: dict, auto_gen_schema_file: str) -> dict:
+    """Update the stack-schema-map.yaml file with the appropriate auto_gen_schema_file location."""
+    random_uuid = str(uuid.uuid4())
+    auto_generated_id = None
+    for version in stack_schema_map:
+        key_found = False
+        for key, value in stack_schema_map[version].items():
+            value_path = resolve_schema_path(value)
+            if value_path == Path(auto_gen_schema_file).resolve() and key not in RESERVED_SCHEMA_NAMES:
+                auto_generated_id = key
+                key_found = True
+                break
+        if key_found is False:
+            if auto_generated_id is None:
+                auto_generated_id = random_uuid
+            stack_schema_map[version][auto_generated_id] = str(auto_gen_schema_file)
+    return stack_schema_map, auto_generated_id, random_uuid
+
+
+def clean_stack_schema_map(stack_schema_map: dict, auto_generated_id: str, random_uuid: str) -> dict:
+    """Clean up the stack-schema-map.yaml file replacing the random UUID with a known key if possible."""
+    for version in stack_schema_map:
+        if random_uuid in stack_schema_map[version]:
+            stack_schema_map[version][auto_generated_id] = stack_schema_map[version].pop(random_uuid)
+    return stack_schema_map
+
+
+def update_auto_generated_schema(index: str, field: str):
```

Comment:
We can probably clean up some of the inline comments in here (probably dont need one for each line).